modules/storages/app/common/storages/peripherals/storage_interaction/storage_queries.rb

```diff
@@ -52,16 +52,23 @@ def download_link_query
       end
     end
 
-    def upload_link_query(finalize_url)
+    def upload_link_query
       case @provider_type
       when ::Storages::Storage::PROVIDER_TYPE_NEXTCLOUD
         retry_with_refreshed_token do |token, with_refreshed_token_proc|
-          ::Storages::Peripherals::StorageInteraction::Nextcloud::UploadLinkQuery.new(
-            base_uri: @uri,
-            token:,
-            retry_proc: with_refreshed_token_proc,
-            finalize_url:
-          )
+          if OpenProject::FeatureDecisions.legacy_upload_preparation_active?
```

Comment:
Maybe we should define a point of time in the future to remove the legacy upload method. A WP or so?

Reply:
I'd couple it to the successful usage of the real direct upload for any amount of time in production.